build/cpp/kvstore/src/main.cc

```diff
@@ -0,0 +1,28 @@
+#include "kvstore_server.h"
+
+DEFINE_string(store, "", "Stores and loads kvstore form .txt files");
+
+kvstore_server::KeyValueStoreServiceImpl *service;
+
+// calls service destructor upon shutdown signal
+void SignalHandler(int signum) {
+  LOG(INFO) << "Handling signal: " << signum;
+  delete service;
+  exit(signum);
+}
+
+// function for termination signals
+void sig() {
+  signal(SIGINT, SignalHandler);
+  signal(SIGTERM, SignalHandler);
+}
+
+int main(int argc, char **argv) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+  google::InitGoogleLogging(argv[0]);
+  service = new kvstore_server::KeyValueStoreServiceImpl(FLAGS_store);
+  std::thread s(sig); // creates thread catching signals
```

Comment:
What is the benefit of making `sig` its own thread?

Reply:
I found the main thread to be halted by the server's Wait() function. By making sig its own thread, the shutdown signal could be caught to call the WriteFile.

Reply:
Ok, that's fine. I was not getting that issue so was just wondering